gschem/src/x_autonumber.c

```diff
@@ -700,7 +700,7 @@ void autonumber_text_autonumber(AUTONUMBER_TEXT *autotext)
 	      /* the beginnig of the current text matches with the searchtext now */
 	      /* strip of the trailing [0-9?] chars and add it too the searchtext */
 	      for (i = strlen (str)-1;
-		   (i >= strlen(searchtext))
+             (i >= (gint) strlen(searchtext))
```

Comment:
Maybe `i` should be `size_t`, since it comes from the return value of `strlen()`.